### ISSUE TITLE
Native backend: RISCV64 support

### ISSUE THREAD

**Issue by adam-7** [gambit/gambit#369]
Please implement RISC-V (64bit) also, as in the longer term its popularity should likely explode.

Also, because of its licensing model, it's a more extensible platform than others, meaning it has a qualitative advantage over AMD/ARM.

Ref.

 * https://www.crowdsupply.com/sifive/hifive-unleashed , https://www.sifive.com/products/hifive-unleashed/ , https://www.sifive.com/documentation/risc-v-core/u54-mc-risc-v-core-ip-manual/

 * https://www.sifive.com/products/hifive1/ (32bit)

 * https://fuse.wikichip.org/news/686/esperanto-exits-stealth-mode-aims-at-ai-with-a-4096-core-7nm-risc-v-monster/ ,  http://eecatalog.com/machine-learning-ai/2018/02/16/qa-with-dave-ditzel-ceo-esperanto-technologies/

If anyone would at any point want to implement native backend support for other remaining production grade architectures (MIPS64, SPARC64, PPC64), then the clarity for how to do so should be fairly easily sourcable from these three already supported architectures.

As a compliment to real hardware, Bochs/QEMU should work well for development/testing environment.

If hardware donations would be welcome please say it & specify which hardware.

**Comment by feeley** [comment]
@abdelq If you are interested in this we should arrange a meeting with Laurent Huberdeau so we can discuss what needs to be done.  One thing for sure that is needed is a RISC-V instruction encoder, similar to ```gsc/_x86.scm``` and ```gsc/_arm.scm```.

**Comment by adam-7** [comment]
@abdelq that is awesome. More urgent may be ARM64 as for instance all phones have that arch. Have you done any progress?

**Comment by abdelq** [comment]
@adam-7 Yes there is progress. Will keep ARM64 in mind as I work on the assembly backend.
